pkg/machinery/nethelpers/hwaddr.go

```diff
@@ -16,12 +20,17 @@ func (addr HardwareAddr) MarshalText() ([]byte, error) {
 
 // UnmarshalText implements text.Unmarshaler interface.
 func (addr *HardwareAddr) UnmarshalText(b []byte) error {
-	mac, err := net.ParseMAC(string(b))
+	rawHex := bytes.ReplaceAll(b, []byte(":"), []byte(""))
```

Comment:
should we add a test to make sure we didn't break anything

Reply:
we already have a test for it (in previous PR): we marshal/unmarshal `LinkStatus` and compare